packages/pipeline-editor/src/PipelineEditorWidget.tsx

```diff
@@ -517,6 +641,7 @@ export class PipelineEditor extends React.Component<
           ] = this.propertiesInfo.parameterDef.current_parameters.include_subdirectories;
 
           this.canvasController.editActionHandler(data);
+          this.setState({ showValidationError: false });
```

Comment:
rather than add this here I think it may make more sense to just set this to false in any call to the editActionHandler that doesn't result in an error (ie clear the error on any "good" pipeline change)